docs/nuvlaedge/installation/requirements.md

```diff
@@ -11,40 +11,90 @@ redirect_from:
+If you are looking at an application with constrained hardware, especially in
+terms of memory size, please get in touch. Some components in
+the [NuvlaEdge architecture](/nuvlaedge/architecture/) are optional and can be
+removed to save memory. Of course, it's a tradeoff.
 
-An remember that you should leave enough memory to host the apps that will be running on the edge device. 
+Please remember that you should leave enough memory to host the apps that will
+be running on the edge device.
 
 ## Software Requirements
 
-Before you can install the NuvlaEdge, please make sure you have:
-
-- [Docker Engine (version 18 or higher)](https://docs.docker.com/install)
-- [Docker Compose (version 1.27.4 or higher)](https://docs.docker.com/compose/install/)
-
+Before you can install the NuvlaEdge, please make sure you have Container
+Orchestration Engine (COE) installed along with the COE specific application
+deployment tools.
+
+For Docker-based edge device use:
+  - [Docker Engine (version 18 or higher)](https://docs.docker.com/engine/install/),
+  - `docker` CLI with `compose` command. Typically, installed along with Docker Engine.
+
+For Kubernetes-based edge device:
+- one of the Kubernetes distributions e.g., [k3s](https://k3s.io/),
+  [k0s](https://k0sproject.io/), [k8s](https://kubernetes.io/docs/setup/production-environment/) 
+  and Container Runtime [CRI](https://kubernetes.io/docs/concepts/architecture/cri/)
+  ([containerd](https://containerd.io/), [CRI-O](https://cri-o.io/) etc.)
+  recommended by the selected Kubernetes distribution.
+- [Helm](https://helm.sh/docs/intro/install/)
+
+### Prerequisites when running on k8s distribution
+
+1. When [k8s](https://kubernetes.io/docs/setup/production-environment/) 
+   distribution of Kubernetes is used, please ensure NuvlaEdge runs on
+   a control-plane node. For that before the installation ensure the
+   selected control-plane node is schedulable (i.e., it needs to be untainted).
+
+2. To collect telemetry data of the containers running on the Kubernetes
+   cluster, NuvlaEdge
+   requires [kubernetes metrics server](https://github.com/kubernetes-sigs/metrics-server)
+   up and running. While e.g., `k3s` or `k0s` distributions provide the _metrics
+   server_ by default, this is not the case for `k8s`. Please install the _metrics
+   server_ following the requirements and procedure provided
+   [here](https://github.com/kubernetes-sigs/metrics-server). In case you do not
+   want to use `--kubelet-insecure-tls` for communication from the _metrics
+   server_ to _kubelet_, you can follow the steps below to provide a valid cert for
+   _kubelet_.
+
+```shell
+# Provide the API server cert/key to the kubelet configuration so that the correct 
+# CA and SANs are present in the server cert/key:
+
+cd /var/lib/kubelet/
+mv kubelet.crt kubelet.crt.bak
+mv kubelet.key kubelet.key.bak
+ln -s /etc/kubernetes/pki/apiserver.crt kubelet.crt
+ln -s /etc/kubernetes/pki/apiserver.key kubelet.key
+```
 
```

Comment:
Should we add a note for k0s that it should be configured to also run under `kubectl` and not only `k0s kubectl` ? e.g. something like this:
https://www.suchcodewow.io/docs/k8s/Kubernetes/k0s
I search for the official k0s link on this subject.

Reply:
Yes, we can. Although what I did, I added this note here

https://github.com/nuvla/documentation/pull/110/files#diff-ac23ccb636b9f5105411eee5b5c54ebecb07d30a7797f57fe4060e519102c192R22-R25

I hope the edge operator who decided to run k0s knows about `k0s kubectl` and with the above note we are simply suggesting to be careful with copy/pasting the command as it might require prepending `k0s`.